### ISSUE TITLE
Environment Files

### ISSUE THREAD

**Issue by piuskamil** [analogjs/analog#868]
### Please provide the environment you discovered this bug in.

NX workspace, analogjs app with ssr

### Which area/package is the issue in?

create-analog

### Description

Analogjs v0.2.30
NX v17.2.8

File replacements don't work. 
I have added env files for development and production.

project.json

```
{
  "name": "test",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "sourceRoot": "./test/src",
  "targets": {
    "build": {
      "executor": "@analogjs/platform:vite",
      "outputs": [
        "{options.outputPath}",
        "{workspaceRoot}/dist/test/.nitro",
        "{workspaceRoot}/dist/test/ssr",
        "{workspaceRoot}/dist/test/analog"
      ],
      "options": {
        "main": "test/src/main.ts",
        "configFile": "test/vite.config.ts",
        "outputPath": "dist/test/client",
        "tsConfig": "test/tsconfig.app.json"
      },
      "defaultConfiguration": "production",
      "configurations": {
        "development": {
          "mode": "development",
          "fileReplacements": [
            {
              "replace": "test/src/environments/environment.ts",
              "with": "test/src/environments/environment.development.ts"
            }
          ]
        },
        "production": {
          "sourcemap": false,
          "mode": "production"
        }
      }
    },
    "serve": {
      "executor": "@analogjs/platform:vite-dev-server",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "test:build",
        "port": 4200
      },
      "configurations": {
        "development": {
          "buildTarget": "test:build:development",
          "hmr": true
        },
        "production": {
          "buildTarget": "test:build:production"
        }
      }
    },
    "extract-i18n": {
      "executor": "@angular-devkit/build-angular:extract-i18n",
      "options": {
        "browserTarget": "test:build"
      }
    },
    "test": {
      ...
    },
    "lint": {
      ...
    }
  },
  "tags": []
}
```

I run commands:

```
nx serve test --configuration=development
nx serve test --configuration=production
```

In both cases, only the production env file is loaded.

No error appeared.

I have added a pure angular app in the same NX project and file replacement work properly.

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

**Comment by luishcastroc** [comment]
ok so it seems Nx Vite uses a plugin for file replacement however when using it i'm getting an error:

```shell
 nx run test:serve:development

failed to load config from /Users/luiscastrocabrera/Projects/analog-test-replacement/test/vite.config.ts

 >  NX   (0 , import_rollup_replace_files.default) is not a function

   Pass --verbose to see the stacktrace.
   ```

So i guess i'll try to file an issue with the Nx team to see if someone can take a look.

**Comment by luishcastroc** [comment]
This is resolved please give it a try, i tried in my blog and it works:

quote from Nx issue: 

This was fixed by [#21077](https://github.com/nrwl/nx/pull/21077), which was released in Nx v17.3. Please migrate your Nx version to at least 17.3.0 to get the fix. Note that the replaceFiles plugin is no longer a default export from that version onwards.
